csrc/scheduler/utils.cpp

```diff
@@ -1181,12 +1172,7 @@ IterDomain* innerMostRootDim(TensorView* tv) {
   IterDomain* inner_most_id = nullptr;
 
   for (auto it = root_domain.rbegin(); it != root_domain.rend(); it++) {
-    // If we're looking at a reduction domain on an input because of
-    // segmentation we don't want to consider those reduction domains as a
-    // vectorization opportunity. If we're looking at a reduction reference
-    // tensor we want to consider the reduction iteration domains as domains we
-    // can vectorize on.
-    if (((*it)->isReduction() && tv->isFusionInput()) || (*it)->isBroadcast()) {
```

Comment:
Why is this changed?

Reply:
This is exactly the purpose of this PR. In the past, this function was called both on the reduction tv and fusion inputs. So we need to special handle reduction tv. Now, this function will not be called on the reduction tv (instead, it will be the producer of the reduction tv), so there is no need to special handle anything.

Reply:
Yes, exactly, and this is why we have all the special handling in these util functions. And these special handlings are hard to maintain and hard to understand, so I want to remove them.